asapdiscovery-data/asapdiscovery/data/state_expanders/stereo_expander.py

```diff
@@ -0,0 +1,45 @@
+from typing import Literal
+
+from asapdiscovery.data.openeye import oechem, oeomega
+from asapdiscovery.data.schema_v2.ligand import Ligand
+from asapdiscovery.data.state_expanders.state_expander import (
+    StateExpanderBase,
+    StateExpanderType,
+    StateExpansion,
+)
+from pydantic import Field
+
+
+class StereoExpander(StateExpanderBase):
+    """
+    Expand a molecule to stereoisomers
+    """
+
+    expander_type: Literal[StateExpanderType.STEREO] = StateExpanderType.STEREO
+    stereo_expand_defined: bool = Field(
+        False,
+        description="Expand stereochemistry at defined centers as well as undefined centers",
+    )
+
+    def _expand(self):
+        flipperOpts = oeomega.OEFlipperOptions()
+        flipperOpts.SetEnumSpecifiedStereo(self.stereo_expand_defined)
+
+        expansions = []
+
+        for ligand in self.input_ligands:
+            expanded_states = []
+
+            oemol = ligand.to_oemol()
+            for enantiomer in oeomega.OEFlipper(oemol, flipperOpts):
+                fmol = oechem.OEMol(enantiomer)
+                # copy the ligand properties over to the new molecule, we may want to have more fine grained control over this
+                # down the track.
+                expanded_states.append(Ligand.from_oemol(fmol, **ligand.dict()))
```

Comment:
Do we have some unique property on the Ligand to show its state expanded and who the parent is? I am assuming if this feeds into the FEC pipeline and at that point it would just be a list of ligands so that would help when collapsing states back down.